test/plugin/test_watch_pods.rb

```diff
@@ -234,30 +234,37 @@ class DefaultPodWatchStrategyTest < WatchTest
       end
     end
 
-    test 'pod watch retries when exceptions are encountered' do
+    test 'pod watch resets watch retry count when exceptions are encountered and connection to k8s API server is re-established' do
```

Comment:
Can we make this a new test and keep the old test that verifies the retry scenario?

Reply:
You're right @qingling128 , there were no tests for raising the `Fluent::UnrecoverableError` after my changes. I wasn't sure how to write such a test with the `minitest`'s stubs, but I think I got it to work using the stubs from `rr`.

Please let me know what you think.

Reply:
Thanks. That looks good to me.